tensorflow_datasets/scripts/random_image_generator.py

```diff
@@ -0,0 +1,132 @@
+
+""" Generate random images that compress better.
+    By replacing the larger images with more compressable equivalents. """
+
+import os
+import tempfile
+import zipfile
+import tarfile
+from PIL import Image
+import numpy as np
+from absl import app
+from absl import flags
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string("dir_path",
+                    "testing/test_data/fake_examples",
+                    "path to the directory which contains files")
+
+def image_process(filepath):
+    """ Generate random images and remove noise of the
+        images to compress better.
+
+          Args:
+           filepath: path of the images to get processed
+
+    """
+    image = np.array(Image.open(filepath))
+    grey = int(hash(filepath) % 255)
+    image = np.ones_like(image)*grey
+    image = Image.fromarray(image)
+    image = image.convert('RGB')
+    image.save(filepath)
+
+
+
+def rewrite_zip(_dir, path):
```

Comment:
`_` means private. And it's good to have explicit names so the user can understand the variable content just from the variable names.
I would rename the args `root_dir, zip_filepath`, and similarly below.